### PR TITLE
Fix integration tests for streaming case

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

vLLM integration tests for the stream case has a few bugs that this addresses

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

File: integration_tests/rest_api_utils.py [hunk 1]
```diff
@@ -987,6 +987,34 @@ def ensure_llm_task_response_is_correct(
         assert re.search(response_text_regex, response["output"]["text"])
 
 
+def ensure_llm_task_stream_response_is_correct(
+    response: str,
+    required_output_fields: Optional[List[str]],
+    response_text_regex: Optional[str],
+):
+    print("raw response", response)
```
Comment (yunfeng-scale) [inline integration_tests/rest_api_utils.py:995]: remove this?

File: integration_tests/rest_api_utils.py [hunk 2]
```diff
@@ -802,7 +802,7 @@ async def create_llm_streaming_task(
         timeout=LONG_NETWORK_TIMEOUT_SEC,
     ) as response:
         assert response.status == 200, (await response.read()).decode()
-        return await response.json()
+        return (await response.read()).decode()
```
Comment (yunfeng-scale) [inline integration_tests/rest_api_utils.py:805]: hmm, how did this work before?

**Review by yunfeng-scale — APPROVED** [review]

File: integration_tests/rest_api_utils.py (hunk 2 above)
Comment (dmchoiboi) [reply integration_tests/rest_api_utils.py:805]: i'm pretty sure it didn't

File: integration_tests/rest_api_utils.py (hunk 1 above)
Comment (dmchoiboi) [reply integration_tests/rest_api_utils.py:995]: we can; it's been helpful for debugging if the tests fail

File: integration_tests/rest_api_utils.py [hunk 3]
```diff
@@ -987,6 +987,34 @@ def ensure_llm_task_response_is_correct(
...
+    #  We want to get a list of dictionaries parsing out the 'data:' field
+    parsed_response = [
+        json.loads(r.split("data: ")[1]) for r in response.split("\n") if "data:" in r.strip()
+    ]
+
+    print("parsed response: ", parsed_response)
+    # if required_output_fields is not None:
+    #     for field in required_output_fields:
+    #         for response in parsed_response:
+    #             assert field in response["output"]
```
Comment (yunfeng-scale) [inline integration_tests/rest_api_utils.py:1007]: should implement these?

Comment (dmchoiboi) [reply integration_tests/rest_api_utils.py:1007]: i'm not really sure what this is supposed to be testing. the response for streaming is different from that for sync so the same checks don't really apply

**Pull request merged**